src/main/java/org/example/dao/UserDaoImpl.java

```diff
@@ -1,77 +1,73 @@
 package org.example.dao;
 
+import org.example.model.ticket.Ticket;
 import org.example.model.user.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
 
-import javax.sql.DataSource;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Timestamp;
 import java.util.UUID;
 
 @Repository
 public class UserDaoImpl implements UserDAO {
+    @Value("${app.user-update-create-ticket.enabled}")
+    private boolean userUpdateCreateEnabled;
 
-    private final DataSource dataSource;
 
-    public UserDaoImpl(DataSource dataSource) {
-        this.dataSource = dataSource;
+    private UserDAO userDAO;
+
+    @Autowired
+    public void setUserDAO(UserDAO userDAO) {
+        this.userDAO = userDAO;
+    }
+
+    private final TicketDAO ticketDao;
```

Comment:
do not inject one DAO class into another DAO class. it is the violation of 'single responsibility principle'

Reply:
Created new TicketService class, moved updateUserAndCreateTicket method into it